Example/AblyChatExample/Mocks/MockSubscriptions.swift

```diff
@@ -0,0 +1,169 @@
+import Ably
+import AblyChat
+
+struct MockMessageSubscription: Sendable, AsyncSequence {
```

Comment:
This could be simplified (not necessarily in terms of amount of code, but I think in readability) by using `AsyncTimerSequence` and `merge` from swift-async-algorithms. Here's an example: 4daab4e03dda4b6bcfb5081e03690c25a1e600cd

Also, all of these `Mock*` types seem to be copied and pasted and basically do the same thing — interleave some timed random elements with some injected ones. Perhaps this could be implemented in a single generic type instead?

Reply:
>This could be simplified (not necessarily in terms of amount of code, but I think in readability) by using AsyncTimerSequence

Interesting approach, but doesn't `typealias Merged = AsyncMerge2Sequence<AsyncMapSequence<AsyncTimerSequence<ContinuousClock>, Message>, AsyncStream<MockMessageSubscription.Element>>` look somewhat confusing (from the first glance at least)? But for the sake of trying new things I have no objections in using it.

>Perhaps this could be implemented in a single generic type instead

I thought about that, but aren't those mocks are there temporary anyways until API implemented? So I went the simplest way possible.

Reply:
https://github.com/ably-labs/ably-chat-swift/pull/34/commits/9dced095d812d26ae2a8bad37d6147ad3963fba9
https://github.com/ably-labs/ably-chat-swift/pull/34/commits/640bcfdfbe853c187ff1ea35830c657201ed4739